lib/web/ui/usercontext_test.go

```diff
@@ -79,55 +78,58 @@ func (s *UserContextSuite) TestNewUserContext(c *check.C) {
 
 	roleSet := []types.Role{role1, role2}
 	userContext, err := NewUserContext(user, roleSet, proto.Features{}, true)
-	c.Assert(err, check.IsNil)
+	require.NoError(t, err)
 
 	allowed := access{true, true, true, true, true}
 	denied := access{false, false, false, false, false}
 
 	// test user name and acl
-	c.Assert(userContext.Name, check.Equals, "root")
-	c.Assert(userContext.ACL.AuthConnectors, check.DeepEquals, allowed)
-	c.Assert(userContext.ACL.TrustedClusters, check.DeepEquals, allowed)
-	c.Assert(userContext.ACL.AppServers, check.DeepEquals, denied)
-	c.Assert(userContext.ACL.DBServers, check.DeepEquals, denied)
-	c.Assert(userContext.ACL.KubeServers, check.DeepEquals, denied)
-	c.Assert(userContext.ACL.Events, check.DeepEquals, denied)
-	c.Assert(userContext.ACL.RecordedSessions, check.DeepEquals, denied)
-	c.Assert(userContext.ACL.Roles, check.DeepEquals, denied)
-	c.Assert(userContext.ACL.Users, check.DeepEquals, denied)
-	c.Assert(userContext.ACL.Tokens, check.DeepEquals, denied)
-	c.Assert(userContext.ACL.Nodes, check.DeepEquals, denied)
-	c.Assert(userContext.ACL.AccessRequests, check.DeepEquals, denied)
-	c.Assert(userContext.ACL.Desktops, check.DeepEquals, allowed)
-	c.Assert(userContext.ACL.WindowsLogins, check.DeepEquals, []string{"a", "b", "d"})
-	c.Assert(userContext.AccessStrategy, check.DeepEquals, accessStrategy{
+	require.Equal(t, userContext.Name, "root")
+	require.Empty(t, cmp.Diff(userContext.ACL.AuthConnectors, allowed))
```

Comment:
Is there a reason we're using `require.Empty( cmp.Diff )` over just `require.Equal`? `require.Equal` also deep compares objects and returns an equally readable output. I think `cmp.Diff` is only needed if you need to ignore certain fields, etc.

Reply:
I see now that `cmp.Diff` avoids using `reflect.DeepEquals`